libimage/image.go

```diff
@@ -279,9 +279,7 @@ func (i *Image) remove(ctx context.Context, rmMap map[string]*RemoveImageReport,
 		return errors.Errorf("cannot remove read-only image %q", i.ID())
 	}
 
-	if i.runtime.eventChannel != nil {
```

Comment:
That is changing the previous semantics. Events are documented to be opt-in and should remain like that. Maybe use a boolean `i.runtime.eventsEnabled`.

Reply:
Same for all other calls of `writeEvent`.